test/odbc/test_pg_datatype_bpchar.cpp

```diff
@@ -0,0 +1,650 @@
+
+  vector<vector<string>> inserted_values = {
+    {PK_VAL, "1", "2", "3"} 
+  };
+
+  vector<vector<string>> updated_values = {
+    {PK_VAL, "a", "b", "c"}, // standard values
+    {PK_VAL, STRING_1, STRING_4000, STRING_20}, // max values
+    {PK_VAL, "", "", ""} // min values
+  };
+
+  vector<tuple<int, int, SQLPOINTER, int, SQLLEN* >> bind_columns{};
+
+  // initialize bind_columns
+  for (int i = 0; i < NUM_COLS; i++) {
+    tuple<int, int, SQLPOINTER, int, SQLLEN*> tuple_to_insert(i+1, SQL_C_CHAR, (SQLPOINTER) &col_results[i], BUFFER_LENGTH, &col_len[i]);
+    bind_columns.push_back(tuple_to_insert);
+  }
+
+  string insert_string = InitializeInsertString(inserted_values);
+
+
+  // Create table
+  odbcHandler.ConnectAndExecQuery(CreateTableStatement(TABLE_NAME, TABLE_COLUMNS_NVARCHAR));
+  odbcHandler.CloseStmt();
+
+  // Insert valid values into the table and assert affected rows
+  odbcHandler.ExecQuery(InsertStatement(TABLE_NAME, insert_string));
+ 
+  rcode = SQLRowCount(odbcHandler.GetStatementHandle(), &affected_rows);
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  ASSERT_EQ(affected_rows, inserted_values.size());
+  
+  odbcHandler.CloseStmt();
+
+  // Select all from the tables and assert that the following attributes of the type is correct:
+  odbcHandler.ExecQuery(SelectStatement(TABLE_NAME, {"*"}, vector<string> {COL_NAMES[0]}));
+
+  // Make sure inserted values are correct
+  ASSERT_NO_FATAL_FAILURE(odbcHandler.BindColumns(bind_columns));
+
+  for (int i = 0; i < inserted_values.size(); ++i) {
+    rcode = SQLFetch(odbcHandler.GetStatementHandle()); // retrieve row-by-row
+    ASSERT_EQ(rcode, SQL_SUCCESS);
+
+    for (int j = 0; j < NUM_COLS; j++) {
+      ASSERT_EQ(string(col_results[j]), inserted_values[i][j]+ operator_multiple(" ",(COL_LENGTH[j]-inserted_values[i][j].size())));
+      ASSERT_EQ(col_len[j], COL_LENGTH[j]);
+    }
+  }
+
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  ASSERT_EQ(rcode, SQL_NO_DATA);
+  odbcHandler.CloseStmt();
+
+  for (int i = 0; i < updated_values.size(); i++) {
+
+    vector<pair<string,string>> update_col;
+    // setup update column
+    for (int j = 0; j <NUM_COLS; j++) {
+      string value = string("'") + updated_values[i][j] + string("'");
+      update_col.push_back(pair<string,string>(COL_NAMES[j], value));
+    }
+
+    odbcHandler.ExecQuery(UpdateTableStatement(TABLE_NAME, update_col, COL_NAMES[0] + "='" + PK_VAL + "'"));
+
+    rcode = SQLRowCount(odbcHandler.GetStatementHandle(), &affected_rows);
+    ASSERT_EQ(rcode, SQL_SUCCESS);
+    ASSERT_EQ(affected_rows, AFFECTED_ROWS_EXPECTED);
+
+    odbcHandler.CloseStmt();
+
+    odbcHandler.ExecQuery(SelectStatement(TABLE_NAME, {"*"}, vector<string> {COL_NAMES[0]}));
+    rcode = SQLFetch(odbcHandler.GetStatementHandle());
+
+    for (int j = 0; j < NUM_COLS; j++) {
+      
+      ASSERT_EQ(string(col_results[j]), updated_values[i][j]+ operator_multiple(" ",(COL_LENGTH[j]-updated_values[i][j].size())));
```

Comment:
```suggestion
    for (int j = 0; j < NUM_COLS; j++) {
      ASSERT_EQ(string(col_results[j]), inserted_values[i][j]+ string(COL_LENGTH[j] - inserted_values[i][j].size(), ' '));
```
Double check brackets again